scribus/plugins/scriptplugin/cmdtext.cpp

```diff
@@ -1367,14 +1367,18 @@ PyObject *scribus_positiontopoint(PyObject* /* self */, PyObject* args)
 		return nullptr;
 	if (!(item->isTextFrame()) && !(item->isPathText()))
 	{
-		PyErr_SetString(WrongFrameTypeError, QObject::tr("Cannot insert text into non-text frame.","python error").toLocal8Bit().constData());
+		PyErr_SetString(WrongFrameTypeError, QObject::tr("Cannot get character positions from a non-text frame.","python error").toLocal8Bit().constData());
 		return nullptr;
 	}
-	if ((pos < 0) || (pos >= static_cast<int>(item->itemText.length())))
+	if ((pos < -1) || (pos > item->lastInFrame() - item->firstInFrame()))
```

Comment:
is `pos == -1` valid?
it's before the first char?

Reply:
ah, no sorry, you're using it below again: it means end of frame...
did you introduce it or did you find it else where in the python API?

personally, i would avoid such behaviors. `pos` should be between 0 (or 1) and length (-1).
i don't think that this shortcut is needed.

Reply:
I was just doing it for consistency with scribus_inserttext(). Removed.